java/meterpreter/stdapi/src/main/java/com/metasploit/meterpreter/stdapi/stdapi_sys_process_execute.java

```diff
@@ -65,10 +65,65 @@ public int execute(Meterpreter meterpreter, TLVPacket request, TLVPacket respons
         return ERROR_SUCCESS;
     }
 
+    // On Windows, Java quote-escapes _some_ arguments (like those with spaces), but doesn't deal correctly with some
+    // edge cases; e.g. empty strings, strings that already have quotes.
+    protected String escapeArg(String arg) {
+        if (System.getProperty("os.name").toLowerCase().contains("windows")) {
```

Comment:
I'm pretty sure this `getProperty` call can't be null - but since the API claims that it's possible:

> the string value of the system property, or null if there is no property with that key.
> https://docs.oracle.com/javase/8/docs/api/index.html?java/lang/System.html

Could we patch this to handle that scenario 😄

Reply:
Done.